src/basic/hostname-util.c

```diff
@@ -191,13 +191,21 @@ bool is_localhost(const char *hostname) {
                 endswith_no_case(hostname, ".localhost.localdomain.");
 }
 
+const char *etc_hostname(void) {
+        return secure_getenv("SYSTEMD_ETC_HOSTNAME") ?: "/etc/hostname";
```

Comment:
The file is also written by firstboot. Maybe better to port this into firstboot.

Reply:
I have mentioned in the commit message that this does not touch firstboot. We do not use it, and I am afraid that if nobody uses it in firstboot, that will break and nobody will see it.

Reply:
Even if your project does not use it, when you write a patch for upstream, please consistently apply the change to all components.

Reply:
yeah, please let's do this comprehensively, in firstboot, too.